dds_web/api/user.py

```diff
@@ -196,46 +196,52 @@ def invite_user(email, new_user_role, project=None, unit=None):
         # Compose and send email
         AddUser.compose_and_send_email_to_user(userobj=new_invite, mail_type="invite", link=link)
 
-        # Append invite to unit if applicable
-        if new_invite.role in ["Unit Admin", "Unit Personnel"]:
-            # TODO Change / move this later. This is just so that we can add an initial Unit Admin.
-            if auth.current_user().role == "Super Admin":
-                if unit:
-                    unit_row = models.Unit.query.filter_by(public_id=unit).one_or_none()
-                    if not unit_row:
-                        raise ddserr.DDSArgumentError(message="Invalid unit publid id.")
-
-                    unit_row.invites.append(new_invite)
-                else:
-                    raise ddserr.DDSArgumentError(message="Cannot invite this user.")
-
-            if "Unit" in auth.current_user().role:
-                # Give new unit user access to all projects of the unit
-                auth.current_user().unit.invites.append(new_invite)
-                for unit_project in auth.current_user().unit.projects:
-                    if unit_project.is_active:
-                        share_project_private_key(
-                            from_user=auth.current_user(),
-                            to_another=new_invite,
-                            from_user_token=dds_web.security.auth.obtain_current_encrypted_token(),
-                            project=unit_project,
-                        )
-
-                if not project:  # specified project is disregarded for unituser invites
-                    msg = f"{str(new_invite)} was successful."
-                else:
-                    msg = f"{str(new_invite)} was successful, but specification for {str(project)} dropped. Unit Users have automatic access to projects of their unit."
+        try:
```

Comment:
> I think the try statement in this part needs to be more granular and split into multiple parts only wrapping the respective share_project_private_key() invocations?
>
>Because if it ever happens that a Unit Admin has not full access to every project of a unit, they now can't invite new Unit Personnel successfully anymore because the try will fail.

True, I'll move the try except to wrap the `share_project_private_key` only.

> Maybe it would be better to include the try statement into share_project_private_key() itself and return True or False as success messages that can be evaluated by the endpoints to generate error messages?

No I don't think we should mess with the share_project_private_key functionality and what it returns. It's good that it raises the key exception, and then we should handle the exception where we need to. Some instances may have different needs.